libs/designsystem/accordion/src/accordion.directive.ts

```diff
@@ -1,8 +1,44 @@
-import { Directive } from '@angular/core';
+import {
+  AfterContentInit,
+  ContentChildren,
+  Directive,
+  Input,
+  OnDestroy,
+  QueryList,
+} from '@angular/core';
+
+import { Subscription } from 'rxjs';
+
+import { AccordionItemComponent } from './accordion-item.component';
 
 @Directive({
   selector: '[kirbyAccordion], kirby-accordion',
   exportAs: 'kirby-accordion',
   standalone: false,
 })
-export class AccordionDirective {}
+export class AccordionDirective implements AfterContentInit, OnDestroy {
+  @Input() headingLevel: number;
+  @ContentChildren(AccordionItemComponent) accordionChildren: QueryList<AccordionItemComponent>;
+
+  private childrenChangesSub?: Subscription;
```

Comment:
The childrenChangesSub should not be needed here, as the QueryList lifecycle is handled by Angular. This means that the on destroy hook can be removed as well.